kernel/kernel-5.10/drivers/usb/core/hub.c

```diff
@@ -5560,9 +5568,19 @@ static void port_event(struct usb_hub *hub, int port1)
 		dev_dbg(&port_dev->dev, "do warm reset\n");
 		if (!udev || !(portstatus & USB_PORT_STAT_CONNECTION)
 				|| udev->state == USB_STATE_NOTATTACHED) {
-			if (hub_port_reset(hub, port1, NULL,
-					HUB_BH_RESET_TIME, true) < 0)
-				hub_port_disable(hub, port1, 1);
+			/* Check for warm reset workaround */
+			if (!warm_reset_workaround) {
```

Comment:
@rene is it possible to detect platform on ARM using DMI functions? in this case you can set this flag from _init function. E.g. like in this patchwork (just did random search) https://patchwork.kernel.org/project/platform-driver-x86/patch/20171005180404.15044-2-hdegoede@redhat.com/

Reply:
good point @rucoder , but in this case it wouldn't help because the Siemens IPC520A uses the default bootloader from Jetpack, so the device is listed by SMBIOS information as `NVIDIA Jetson Xavier NX Developer Kit`, exactly the same as the developer kit board (which doesn't require this workaround). On EVE, I'm using a trick (looking to the asset tag) to differentiate the real developer kit from this device, but in the future (when we get the interactive installer), the correct device model will be set at installation phase, so the idea is to keep enabling this workaround from the kernel command line....

Reply:
yes, in our device tree files, no problem. But during the boot we rely on the device tree information from the bootloader, then based on this information we can load the correct device tree from our kernel (https://github.com/lf-edge/eve/blob/master/pkg/grub/rootfs.cfg#L292). At first glance I wouldn't like to diverge the SMBIOS model from the original one (from the bootloader). Since we need to pass other specific parameters through the cmdline for these devices (like `pci=noaer`, `fbcon=map:0` for this Siemens device) I don't see any big problem in use the module parameter to enable/disable the workaround...

Reply:
ok, got it